text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,550 @@
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+Today, we are launching a smaller version of the AWS CDK v2 framework library,
+`aws-cdk-lib`. And, a change in the AWS CDK v2 CLI to support this smaller
+framework library. Before today’s release, the unpacked size of this package was
+234 MB. This large size prevented many customers from being able to use it, and
+caused performance problems for others. It also prevented the CDK team from
+releasing certain new features. Now, the size of the package is less than 100
+MB. This smaller version also has a mechanism for adding new Lambda Layer
+dependencies. Because of this launch, the AWS CDK framework will be able to add
+support for more versions of kubectl, and awscliv2.
+
+### Why should I use this feature?
+
+You should upgrade your dependencies on `aws-cdk-lib` to get the benefits of a
+smaller package size. If you use `aws-cdk-lib` in a Lambda function, now you can
+include more code and dependencies before hitting the Lambda function code size
+limit of 250 MB. If you use `aws-cdk-lib` in your CI/CD pipelines and are
+frequently downloading it with a fresh npm install, you will see performance
+improvements in the download times.
+
+## Internal FAQ
+
+### Why is a large package a problem? Why are we doing this?
+
+There are many reasons this is an issue for customers. The following items need
+to be addressed.
+
+1. The package managers that `aws-cdk-lib` is published on have real or
+   recommended filesize limits. For example:
+    1. The v2 release currently gets this warning when publishing for Go. `Go:
+       remote: warning: File awscdk/jsii/aws-cdk-lib-2.17.0.tgz is 55.42 MB;
+       this is larger than GitHub's recommended maximum file size of 50.00 MB`
+    2. NPM itself does not have a package size limit, but other services that
+       host NPM packages do. For example, [Azure
+       Artifacts](https://docs.microsoft.com/en-us/azure/devops/artifacts/reference/limits?view=azure-devops)
+       has a limit of 500 MB.
+    3. PyPi has an unofficially documented limit of 60 MB.
+2. A large download size might be problematic for customers with weak internet
+   connections or limited bandwidth.
+3. Customers who download and install `aws-cdk-lib` on every CI/CD pipeline run
+   may be frustrated to spend resources frequently downloading a large package.
+4. AWS Lambda limits the size of your function code to 250 MB.
+5. The AWS Lambda limit for all uploaded assets for a Function is 75 GB. If a
+   customer is using Lambda Versions, and wants to maintain a Version for every
+   version they have published, they might end up hitting this limit because of
+   the large size of `aws-cdk-lib`.
+6. The developer experience of using `aws-cdk-lib` in an IDE might be slower in
```

Comment:
Pretty sure this one is not going to be true. If this is the case, it will be because of the large swathes of tiny `.d.ts` files, not a couple of large `.zip` files.

Reply:
Hmm okay, I will just remove this then. I think we have enough other data for why we need to do this project.